src/SimplifyCorrelatedDifferences.cpp

```diff
@@ -57,7 +60,11 @@ class SimplifyCorrelatedDifferences : public IRMutator {
         result = mutate(result);
 
         for (auto it = frames.rbegin(); it != frames.rend(); it++) {
-            result = LetStmtOrLet::make(it->op->name, it->op->value, result);
```

Comment:
The changes in this file are because sizing the global pool well requires careful handling of correlated difference expressions. I discovered that the pass to simplify correlated difference expressions was not handling let values.